components/org.wso2.carbon.identity.application.authentication.handler.identifier/src/main/java/org/wso2/carbon/identity/application/authentication/handler/identifier/IdentifierHandler.java

```diff
@@ -441,6 +476,33 @@ protected boolean retryAuthenticationEnabled() {
         return true;
     }
 
+    protected AuthenticatorFlowStatus executeAutoLoginFlow(AuthenticationContext context, Cookie autoLoginCookie)
+            throws AuthenticationFailedException {
+
+        String decodedValue = new String(Base64.getDecoder().decode(autoLoginCookie.getValue()));
+        JSONObject cookieValueJSON = Utilities.transformToJSON(decodedValue);
+        String signature = (String) cookieValueJSON.get(AutoLoginConstant.SIGNATURE);
+        String content = (String) cookieValueJSON.get(AutoLoginConstant.CONTENT);
+        JSONObject contentJSON = Utilities.transformToJSON(content);
+        Utilities.validateAutoLoginCookie(context, getAuthenticatorConfig(), content, signature);
+
+        String usernameInCookie = (String) contentJSON.get(AutoLoginConstant.USERNAME);
+
+        if (log.isDebugEnabled()) {
+            log.debug("Started executing Auto Login from Cookie flow.");
```

Comment:
can we add some context here? userId, tenant,etc?

Reply:
In here we only have the username sent in the cookie. Logging human-readable user identifiers are discouraged due to privacy concerns. And tenant information will be available in the log data. If needed we can log the cookie content here.

Reply:
let's add cookie content then.

Reply:
let's not log the cookie content as it is like a secret we trust to log in the user without prompting the password.